lib/web_ui/dev/driver_manager.dart

```diff
@@ -0,0 +1,145 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io' as io;
+
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as pathlib;
+import 'package:web_driver_installer/chrome_driver_installer.dart';
+import 'package:web_driver_installer/safari_driver_runner.dart';
+
+import 'chrome_installer.dart';
+import 'common.dart';
+import 'environment.dart';
+import 'utils.dart';
+
+/// [DriverManager] implementation for Chrome.
+///
+/// This manager can be used for both MacOS and Linux.
+class ChromeDriverManager extends DriverManager {
+  ChromeDriverManager(String browser) : super(browser);
+
+  Future<void> _installDriver() async {
+    if (_browserDriverDir.existsSync()) {
+      _browserDriverDir.deleteSync(recursive: true);
+    }
+
+    _browserDriverDir.createSync(recursive: true);
+    temporaryDirectories.add(_drivers);
+
+    io.Directory temp = io.Directory.current;
+    io.Directory.current = _browserDriverDir;
+
+    // TODO(nurhan): https://github.com/flutter/flutter/issues/53179
+    final String chromeDriverVersion = await queryChromeDriverVersion();
+    ChromeDriverInstaller chromeDriverInstaller =
+        ChromeDriverInstaller.withVersion(chromeDriverVersion);
+    await chromeDriverInstaller.install(alwaysInstall: true);
+    io.Directory.current = temp;
```

Comment:
If a statement between line 32 and this line throws an exception, the current directory will be left pointing to `_browserDriverDir`. Consider wrapping the code in `try/finally` and restoring the current directory in the `finally` block. However, an even better design would be to not rely on mutable global state. For example, is it possible to pass the current directory to whoever needs it?

Reply:
I did the first one, the second one is a good suggestion yet it requires me to change the other repo.